sklearn/linear_model/tests/test_ridge.py

```diff
@@ -2084,3 +2086,37 @@ def test_ridge_sample_weight_consistency(
     assert_allclose(reg1.coef_, reg2.coef_)
     if fit_intercept:
         assert_allclose(reg1.intercept_, reg2.intercept_)
+
+
+# Metadata Routing Tests
+# ======================
+
+
+@pytest.mark.usefixtures("enable_slep006")
+@pytest.mark.parametrize("metaestimator", [RidgeCV, RidgeClassifierCV])
+def test_metadata_routing_with_default_scoring(metaestimator):
+    """Test that `RidgeCV` or `RidgeClassifierCV` with default `scoring`
+    argument (`None`), don't enter into `RecursionError` when metadata is routed.
+    """
+    X = np.array([[0, 1], [2, 2], [4, 6], [9, 0], [2, 4]])
+    y = [1, 2, 3, 4, 5]
+
+    pipe = SimplePipeline(
+        [
+            ConsumingTransformer()
+            .set_fit_request(sample_weight=True)
+            .set_transform_request(sample_weight=True),
+            ConsumingTransformer()
+            .set_fit_request(sample_weight=True)
+            .set_transform_request(sample_weight=True),
+            metaestimator().set_fit_request(sample_weight=True),
+        ]
+    )
+
+    params = {"sample_weight": [1, 1, 1, 1, 1]}
+
+    pipe.fit(X, y, **params)
```

Comment:
The actual minimal reproducible would be:

```py
    metaestimator().get_metadata_routing()
```

So we can remove the rest.

Reply:
Yes, this is another (and more basic) way to trigger the error.